contracts/collections/sg721-base/src/contract.rs

```diff
@@ -1,275 +1,172 @@
-                    cw721_base::ExecuteMsg::UpdateOwnership(msg),
-                )
-                .map_err(|e| ContractError::OwnershipUpdateError {
-                    error: e.to_string(),
-                }),
+                .execute(deps, &env, &info, msg.into())
+                .map_err(|e| e.into()),
         }
     }
 
+    #[allow(deprecated)]
     pub fn update_collection_info(
         &self,
         deps: DepsMut,
         env: Env,
         info: MessageInfo,
         collection_msg: UpdateCollectionInfoMsg<RoyaltyInfoResponse>,
-    ) -> Result<Response, ContractError> {
-        let mut collection = self.collection_info.load(deps.storage)?;
+    ) -> Result<Response<TCustomResponseMsg>, ContractError> {
+        let collection_info = self
+            .parent
+            .query_collection_info_and_extension(deps.as_ref())?;
 
         if self.frozen_collection_info.load(deps.storage)? {
             return Err(ContractError::CollectionInfoFrozen {});
         }
 
-        // only creator can update collection info
-        if collection.creator != info.sender {
-            return Err(ContractError::Unauthorized {});
-        }
+        // in this contract, extension is always present, so unwrap is safe
+        let collection_extension = collection_info.extension.unwrap();
 
-        if let Some(new_creator) = collection_msg.creator {
-            deps.api.addr_validate(&new_creator)?;
-            collection.creator = new_creator;
-        }
-
-        collection.description = collection_msg
-            .description
-            .unwrap_or_else(|| collection.description.to_string());
-        if collection.description.len() > MAX_DESCRIPTION_LENGTH as usize {
+        if collection_extension.description.len() > MAX_COLLECTION_DESCRIPTION_LENGTH as usize {
             return Err(ContractError::DescriptionTooLong {});
         }
 
-        collection.image = collection_msg
-            .image
-            .unwrap_or_else(|| collection.image.to_string());
-        Url::parse(&collection.image)?;
-
-        collection.external_link = collection_msg
-            .external_link
-            .unwrap_or_else(|| collection.external_link.as_ref().map(|s| s.to_string()));
-        if collection.external_link.as_ref().is_some() {
-            Url::parse(collection.external_link.as_ref().unwrap())?;
-        }
-
-        collection.explicit_content = collection_msg.explicit_content;
-
-        if let Some(Some(new_royalty_info_response)) = collection_msg.royalty_info {
+        if let Some(Some(_)) = collection_msg.royalty_info {
             let last_royalty_update = self.royalty_updated_at.load(deps.storage)?;
             if last_royalty_update.plus_seconds(24 * 60 * 60) > env.block.time {
                 return Err(ContractError::InvalidRoyalties(
                     "Royalties can only be updated once per day".to_string(),
                 ));
             }
 
-            let new_royalty_info = RoyaltyInfo {
```

Comment:
How come we can delete all this royalty calculation?

Reply:
That is imo the cool part about it, it is moved to cw721. RoyalInfoMsg implements StateFactory where in `create()` it also does `validate()`.

Check in other PR my comment here: https://github.com/CosmWasm/cw-nfts/pull/156/files#r1541437296